### PR TITLE
Add csrf explanation

### PR DESCRIPTION
**Pull request by ProbablePrime**
Tried to keep it brief but verbose enought to explain everything

### CODE REVIEW

File: src/rest/template.pug [hunk 1]
```diff
@@ -30,6 +30,14 @@ include ./resource.pug
...
                         td= bucket.max
                         td= bucket.interval/1000
+        h2(id='csrf') CSRF Tokens
+        p.
+            Beam's REST API requires the use of CSRF Tokens when using cookie-base authentication.
+        p.
+            CSRF tokens are alphanumeric strings issued by the server and are included in the X-CSRF-Token header in responses. They are used to authorize requests and are bound a specific session cookie. They are invalidated after the session expires and may be invalidated at any point prior to the session expiration.
+
+        p.
+            CSRF tokens must be passed to the server in the X-CSRF-Token header when making requests which use cookie-based authentication. Failing to pass a valid CSRF token will cause the request to return in a #[code 461 CSRF token invalid] response. The server will include an X-CSRF-Token header in the response, which contains a CSRF token that the client can use to retry the request.
```
Comment (connor4312) [inline src/rest/template.pug]: Caps, `461 CSRF Token Invalid`

**Comment by connor4312** [comment]
Looks good. Maybe we can include an example (in a spoiler-style expander)? Like

```bash
$ # Without passing a CSRF token:
$ http get https://beam.pro/api/v1/chats/129
HTTP/1.1 461 CSRF token invalid
Content-Type: application/json
X-CSRF-Token: dca5130a5a6390bd83c077a902e1edf8d052bcc8

{
    "error": "Invalid CSRF header",
    "statusCode": 401
}

$ # Now we have a CSRF token to use:
$ http get https://beam.pro/api/v1/chats/129 X-CSRF-Token: dca5130a5a6390bd83c077a902e1edf8d052bcc8
HTTP/1.1 200 OK
...
```

optionally translating to the equivalent cURL

**Comment by Doridian** [comment]
Make sure to describe the new JSON object as I just fixed it up to be like exactly how it will look. No more `statusCode: 401`

**Comment by ProbablePrime** [comment]
Added an example: @connor4312

**Comment by connor4312** [comment]
Neat! I might poke at that with my magic CSS wand later today :)

**Comment by SimonSchick** [comment]
lgtm in current state

**Pull request merged**